benchmarks/instruction_counts/execution/runner.py

```diff
@@ -0,0 +1,240 @@
+import math
+import multiprocessing
+import subprocess
+import textwrap
+import threading
+import time
+from typing import Dict, List, Optional, Set, Tuple, Union
+
+from execution.work import PYTHON_CMD, SHELL, InProgress, WorkOrder
+from worker.main import WorkerFailure, WorkerOutput
+
+
+CPU_COUNT: int = multiprocessing.cpu_count()
+
+
+class WorkerFailed(Exception):
+    """Raised in the main process when a worker failure is detected."""
+    def __init__(self, wrapped_trace: Optional[str] = None) -> None:
+        self.wrapped_trace: Optional[str] = wrapped_trace
+        super().__init__()
+
+
+class CorePool:
+    def __init__(self, min_core_id: int, max_core_id: int) -> None:
+        assert min_core_id >= 0
```

Comment:
Love the handling of assertions and types here! Great!!

Reply:
Thanks! One think that MyPy does that's pretty cool is type narrowing. So if you have `x: Optional[int]` and you `assert x is not None`, for the rest of the scope x is treated as `int` rather than `Optional[int]`. (And same for `Union`, etc.) Not directly applicable here, but I bring it up because just making things mypy-strict compliant has made me subconsciously start adding a bunch more asserts.